frontend/packages/client/src/components/Community/StrategyEditorModal/StrategyInformationForm.js

```diff
@@ -38,31 +86,101 @@ export default function StrategyInformationForm({
   });
 
   const { isDirty, isSubmitting, errors, isValid } = formState;
+  const selectedContract = useWatch({ control, name: 'contract' });
+
+  // On mount to see if a dropdown should be shown based on the type of strategy and if there's
+  // data in the dropdown
+  useEffect(() => {
+    const { dropDownData = [] } = getDropDownData({
+      strategyKey: strategy?.key,
+    });
+
+    if (dropDownData.length > 0) {
+      setIsDropDownVisible(true);
+      setContractOption(dropDownData);
+    }
+  }, []);
+
+  useEffect(() => {
+    // When there is no dropdown, the fields should be filled on component mount itself
+    switch (strategy.key) {
+      case 'staked-token-weighted-default': {
+        const contract = getContractsAndPathsDataWithKeyValue(
+          'contractName',
+          'FlowToken'
+        );
+        setFormDetails(contract);
+        break;
+      }
+      case 'float-nfts': {
+        const contract = getContractsAndPathsDataWithKeyValue(
+          'contractName',
+          'FLOAT'
+        );
+        setFormDetails(contract);
+        break;
+      }
+      default:
+        break;
+    }
+
+    // When there's a dropdown and the user has selected one of the contract.
+    if (selectedContract) {
+      const contract = getContractsAndPathsDataWithKeyValue(
+        'contractName',
+        selectedContract
+      );
+      setFormDetails(contract);
+    }
+  }, [strategy, selectedContract]);
+
+  const setFormDetails = ({ contractName, contractAddress, publicPath }) => {
+    setValue('name', contractName);
+    setValue('addr', contractAddress);
+    setValue('publicPath', publicPath);
+  };
 
   return (
     <form onSubmit={handleSubmit(onSubmit)}>
       <div className="columns is-flex-direction-column is-mobile m-0">
-        <div className="small-text" style={{ color: '#757575' }}>
+        <div className="small-text has-text-grey">
           Need help finding this information?{' '}
           <a
-            href="https://docs.cast.fyi"
+            href="https://dapper-collectives-1.gitbook.io/cast-docs/working-with-contracts-and-paths"
```

Comment:
```suggestion
            href="https://docs.cast.fyi/cast-docs/working-with-contracts-and-paths"
```

Reply:
there is a 302 redirect in place for docs.cast.fyi to goto the gitbook link.
I think we should keep using that redirect specially we already plan to change the dapper-collective subdomain (see [issue](https://github.com/onflow/CAST/issues/12))

Reply:
The motive was to redirect the user to the particular "Working with Contracts and Paths" Page.
However if the redirect is happening, it's not taking care of the particular route we want the user to see. It'll redirect to the main page of gitbook.
I would request you to visit `https://docs.cast.fyi/cast-docs/working-with-contracts-and-paths` @vishalchangrani

Reply:
ah ok - didn't realize that. The 302 hasn't been set correctly I guess.
ok, lets keep it the way you have it for now - and then when we change the subdomain on the gitbook as part of that issue, just remember to change this link.